### PR TITLE
feat: adds AWS_DEFAULT_REGION to barman env

### LINKED ISSUE #202: Scaleway S3 requires a region for backups and wal archive
When using the Scaleway S3 compatible buckets, wal archive fails with the following error:

```
ERROR: Barman cloud WAL archive check exception: An error occurred (400) when calling the HeadBucket operation: Bad Request","pipe":"stderr","logging_pod":"cluster-example-1"}
{"level":"error","ts":1653462323.3326237,"logger":"wal-archive","msg":"Error invoking barman-cloud-check-wal-archive","logging_pod":"cluster-example-1","walName":"000000010000000000000001","currentPrimary":"cluster-example-1","targetPrimary":"cluster-example-1","options":["--endpoint-url","https://s3.fr-par.scw.cloud/","--cloud-provider","aws-s3","s3://mybucket","cluster-example"],"exitCode":-1,"error":"exit status 4","stacktrace":"[github.com/cloudnative-pg/cloudnative-pg/pkg/management/barman/archiver.(*WALArchiver).CheckWalArchive](http://github.com/cloudnative-pg/cloudnative-pg/pkg/management/barman/archiver.(*WALArchiver).CheckWalArchive)\n\tpkg/management/barman/archiver/archiver.go:227\[ngithub.com/cloudnative-pg/cloudnative-pg/internal/cmd/manager/walarchive.run](http://ngithub.com/cloudnative-pg/cloudnative-pg/internal/cmd/manager/walarchive.run)\n\tinternal/cmd/manager/walarchive/cmd.go:176\[ngithub.com/cloudnative-pg/cloudnative-pg/internal/cmd/manager/walarchive.NewCmd.func1](http://ngithub.com/cloudnative-pg/cloudnative-pg/internal/cmd/manager/walarchive.NewCmd.func1)\n\tinternal/cmd/manager/walarchive/cmd.go:79\[ngithub.com/spf13/cobra.(*Command).execute](http://ngithub.com/spf13/cobra.(*Command).execute)\n\tpkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856\[ngithub.com/spf13/cobra.(*Command).ExecuteC](http://ngithub.com/spf13/cobra.(*Command).ExecuteC)\n\tpkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974\[ngithub.com/spf13/cobra.(*Command).Execute](http://ngithub.com/spf13/cobra.(*Command).Execute)\n\tpkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902\nmain.main\n\tcmd/manager/main.go:61\nruntime.main\n\t/opt/hostedtoolcache/go/1.17.8/x64/src/runtime/proc.go:255"}
```

The failure of the `barman-cloud-check-wal-archive` is related to the fact that the command doesn't know the `AWS_REGION`, which should be set to `fr-par` in the example.

Checking manually with barman-cli-cloud only works when `~/.aws/config` is set to

```
[default]
region = fr-par
```

or when passing `AWS_DEFAULT_REGION=fr-par` as an env variable to the command.

### PR DESCRIPTION
**Pull request by armru**
Closes #202

### CODE REVIEW

**Comment by armru** [comment]
do not squash

File: pkg/management/barman/credentials/env.go [hunk 1]
```diff
@@ -103,55 +103,57 @@ func envSetAWSCredentials(
...
-	if !ok {
-		return nil, fmt.Errorf("missing key inside access key ID secret")
+	accessKeyID, acessKeyErr := extractValueFromSecret(
+		ctx,
+		c,
+		s3credentials.AccessKeyIDReference,
+		namespace,
+	)
+	if acessKeyErr != nil {
+		return nil, acessKeyErr
```
Comment (fcanovai) [inline pkg/management/barman/credentials/env.go:117]:
```suggestion
	accessKeyID, accessKeyErr := extractValueFromSecret(
		ctx,
		c,
		s3credentials.AccessKeyIDReference,
		namespace,
	)
	if accessKeyErr != nil {
		return nil, accessKeyErr
```

File: pkg/management/barman/credentials/env.go [hunk 2]
```diff
@@ -103,55 +103,57 @@ func envSetAWSCredentials(
...
+		region, regionErr := extractValueFromSecret(
+			ctx,
+			c,
+			s3credentials.RegionReference,
+			namespace,
+		)
+		if regionErr != nil {
+			return nil, regionErr
+		}
+		env = append(env, fmt.Sprintf("AWS_DEFAULT_REGION==%s", region))
```
Comment (fcanovai) [inline pkg/management/barman/credentials/env.go:144]:
```suggestion
		env = append(env, fmt.Sprintf("AWS_DEFAULT_REGION=%s", region))
```

File: api/v1/cluster_types.go [hunk 3]
```diff
@@ -1037,6 +1037,9 @@ type S3Credentials struct {
 	// The reference to the secret access key
 	SecretAccessKeyReference *SecretKeySelector `json:"secretAccessKey,omitempty"`
 
+	// The reference to the secret containing the region name
+	RegionReference *SecretKeySelector `json:"region,omitempty"`
+
```
Comment (fcanovai) [inline api/v1/cluster_types.go:1042]: I think it's missing a `make generate`, I do not see anything in the deepcopy file related to this

**Review by fcanovai — CHANGES_REQUESTED** [review]

**Review by fcanovai — APPROVED** [review]

**Review by sxd — APPROVED** [review]

**Pull request merged**